bcda/monitoring/monitoring.go

```diff
@@ -26,6 +27,27 @@ func (a apm) Start(msg string, w http.ResponseWriter, r *http.Request) *newrelic
 	return nil
 }
 
+func (a apm) NewTransaction(name string, ctx context.Context) (*newrelic.Transaction, context.Context) {
+	if a.App != nil {
+		txn := a.App.StartTransaction(name)          // transaction trace
+		newRelicCtx := newrelic.NewContext(ctx, txn) // parent context
+		return txn, newRelicCtx
+	}
+	return nil, nil
+}
+
+func NewSpan(parentCtx context.Context, name string) (close func()) {
+	txn := newrelic.FromContext(parentCtx)
+	if txn == nil {
+		log.API.Warn("No transaction found. Cannot create child.")
+	}
+	segment := txn.StartSegment(name)
```

Comment:
If use an integration package (like this one: https://pkg.go.dev/github.com/newrelic/go-agent/v3/integrations/nrpgx), then we don't have to create a [segment manually each time](https://github.com/newrelic/go-agent/blob/master/GUIDE.md#datastore-segments)e, which is a cool functionality!